android/expoview/src/main/java/host/exp/exponent/utils/ScopedContext.java

```diff
@@ -57,8 +59,11 @@
   private File mCacheDir;
   private ScopedApplicationContext mScopedApplicationContext;
 
-  public ScopedContext(final Context context, final String scope) {
+  public ScopedContext(final Context context, final ExperienceKey experienceKey) throws UnsupportedEncodingException {
     super(context);
+
+    String scope = experienceKey.getUrlEncodedScopeKey();
```

Comment:
This seems like a particularly dangerous change given how many modules access shared preferences through this class. We may need to use the legacy stable ID here as well if any of those modules access this class in standalone apps (which I'm not seeing any reason they wouldn't, but I might be blind).

Reply:
We almost never want to use the stable legacy ID field. The scope key will be the correct value here.

Reply:
We almost never want to use the stable legacy ID field. The scope key will be the correct value here.

Reply:
The actual value of the scope key is different from the stable legacy ID, even for apps that existed before any of these changes, correct? I'm just saying that regardless of which value is ideologically correct to use here, I think it's quite possible that changing this value will break things in standalone apps, for modules that access SharedPreferences through this ScopedContext (which there are a number of).

Reply:
No, the scope key for old projects is equal to their original full name i.e. the stable legacy ID.

The scope key for new projects will be a separate, dedicated value.

Reply:
Oh 🤦 well in that case I think basically all my comments can be disregarded. Will edit as such. Sorry for the confusion.